lib/stub_alloc_pages.c

```diff
@@ -42,6 +42,15 @@
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 
+/* Expose the address of the bigarray (in other words, leak the pointer).
+  */
+CAMLprim value
+caml_address(value t)
+{
+  long data = (long)Caml_ba_data_val(t) ;
+  return Val_long(data);
```

Comment:
Ah, thank you for your comments, this stuff is super confusing to me. I haven't been able to find such a platform, but I'll change the PR. Do you know why there is a `uintnat` when the `nativeint` exposed in OCaml seems to act as a signed integer?